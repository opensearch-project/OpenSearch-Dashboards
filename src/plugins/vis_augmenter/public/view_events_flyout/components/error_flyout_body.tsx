/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlyoutBody, EuiFlexGroup, EuiFlexItem, EuiCallOut } from '@elastic/eui';

interface Props {
  errorMessage: string;
}

export function ErrorFlyoutBody(props: Props) {
  return (
    <EuiFlyoutBody>
      <EuiFlexGroup>
        <EuiFlexItem grow={true}>
          <EuiCallOut color="danger" iconType="alert" data-test-subj="errorCallOut">
            {props.errorMessage}
          </EuiCallOut>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlyoutBody>
  );
}
