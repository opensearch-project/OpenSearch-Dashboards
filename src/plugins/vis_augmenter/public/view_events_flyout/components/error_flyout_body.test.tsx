/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ErrorFlyoutBody } from './error_flyout_body';

describe('<ErrorFlyoutBody/>', () => {
  const errorMsg = 'oh no an error!';
  it('shows error message', async () => {
    const { getByText } = render(<ErrorFlyoutBody errorMessage={errorMsg} />);
    expect(getByText(errorMsg)).toBeInTheDocument();
  });
  it('renders component', async () => {
    const { container, getByTestId } = render(<ErrorFlyoutBody errorMessage={errorMsg} />);
    expect(getByTestId('errorCallOut')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
