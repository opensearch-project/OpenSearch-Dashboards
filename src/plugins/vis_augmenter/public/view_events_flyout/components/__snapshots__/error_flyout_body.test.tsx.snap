// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorFlyoutBody/> renders component 1`] = `
<div>
  <div
    class="euiFlyoutBody"
  >
    <div
      class="euiFlyoutBody__overflow"
      tabindex="0"
    >
      <div
        class="euiFlyoutBody__overflowContent"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiCallOut euiCallOut--danger"
              data-test-subj="errorCallOut"
            >
              <div
                class="euiText euiText--small"
              >
                <div
                  class="euiTextColor euiTextColor--default"
                >
                  oh no an error!
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
